flume-ng-doc/sphinx/FlumeUserGuide.rst

```diff
@@ -2166,10 +2166,29 @@ Property Name            Default  Description
 **kite.dataset.uri**     --       URI of the dataset to open
 kite.repo.uri            --       URI of the repository to open
                                   (deprecated; use kite.dataset.uri instead)
+kite.dataset.namespace   --       Namespace of the Dataset where records will be written
+                                  (deprecated; use kite.dataset.uri instead)
 kite.dataset.name        --       Name of the Dataset where records will be written
                                   (deprecated; use kite.dataset.uri instead)
 kite.batchSize           100      Number of records to process in each batch
 kite.rollInterval        30       Maximum wait time (seconds) before data files are released
+kite.commitOnBatch       true     If true, the file will be synced and the Flume transaction
+                                  will be commited on each batch of ``kite.batchSize`` records.
+                                  Parquet doesn't support sync, so this setting will be ignored
+                                  and the behavior will be the same as if this was ``false``.
```

Comment:
The only problem with that is we can't validate the parameter in the config because the target dataset may not yet exist. That means we have to delay until the sink has been started to detect the configuration error. I could change the parameter to kite.avro.commitOnBatch. The only issue is how to what effect this should, if any, have on HBase datasets or future dataset implementations we may add in the future.